File: rup-projects-core/src/main/java/com/core/rupprojectscore/entity/UseCase.java
package com.core.rupprojectscore.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "use_case")
public class UseCase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, length = 25)
    private String name;

    @Column(length = 200)
    private String description;

    @Column
    private Integer priority = 0;

}

File: web-app/src/app/models/use-case.ts
export class UseCase {

  constructor(public  id: number,
              public  name: string,
              public  description: string,
              public priority: number) {
  }

  static build(useCase: UseCase): UseCase {
    return new UseCase(useCase.id, useCase.name, useCase.description, useCase.priority);
  }

}
